### PR TITLE
Add stateless_categorical for TensorFlow frontend

### LINKED ISSUE #17747: stateless_categorical
Add stateless_categorical for TensorFlow frontend

### PR DESCRIPTION
**Pull request by eekamal**
fixes #17747

### CODE REVIEW

File: ivy/functional/frontends/tensorflow/random.py [hunk 1]
```diff
@@ -83,3 +83,16 @@ def stateless_poisson(shape, seed, lam, dtype=ivy.int32, name=None):
...
+@to_ivy_arrays_and_back
+def stateless_categorical(logits, num_samples, seed,dtype=tf.dtypes.int64,name=None):
+    ivy.seed(seed)
+    probabilities = ivy.softmax(logits)
+    samples = ivy.empty((num_samples,), dtype=ivy.int64)
+    for i in range(num_samples):
+        random_number = ivy.random_uniform(shape=(1,), low=0.0, high=1.0)
+        cumulative_probs = ivy.cumsum(probabilities)
+        sample = ivy.argmax(cumulative_probs > random_number)
+        samples = ivy.assign(samples, sample, indices=i)
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/tensorflow/random.py:97]: There is no ivy.assign method, you will have to find an alternative

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
Hey @eekamal, thanks for making the pull request! I've left a couple of comments, please could you make the changes accordingly. Also please can you ensure that the test you wrote is passing? You can run it with the command pytest ivy_tests/test_ivy/test_frontends/test_tensorflow/test_random.py::test_ivy_stateless_categorical 🙂

**Comment by eekamal** [comment]
I tried but I really need more guidance please @Sam-Armstrong

**Comment by Sam-Armstrong** [comment]
@eekamal I'd recommend reading through these two sections of the docs to help you out:
https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.html#ivy-frontends
https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends_tests.html

If you have any specific questions, feel free to let me know either here or on discord 😁

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_random.py [hunk 2]
```diff
@@ -357,3 +357,58 @@ def test_tensorflow_gamma(
...
+    seed = draw(
+        helpers.dtype_and_values(
+            available_dtypes=("int64", "int32"), min_value=0, max_value=10, shape=[2]
+        )
+    )
+    return logits, num_samples, seed, dtype
+
+
+@handle_frontend_test(
+    fn_tree="ivy.random.stateless_categorical",
```
Comment (Sam-Armstrong) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_random.py:390]:
```suggestion
    fn_tree="tensorflow.random.stateless_categorical",
```

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
@eekamal Sorry its taken a while for me to look at this, I've been really busy. I've suggested a change here which allows the tests to run, but looks like the test cases are failing. Please could you try to debug the problem? Thanks 😊

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
@eekamal the tests are still failing for me. Have you ran them with pytest ivy_tests/test_ivy/test_frontends/test_tensorflow/test_random.py::test_ivy_stateless_categorical ?

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
